compiler/noirc_frontend/src/hir/comptime/interpreter/builtin.rs

```diff
@@ -487,20 +492,101 @@ fn quoted_as_type(
     Ok(Value::Type(typ))
 }
 
+// fn as_array(self) -> Option<(Type, Type)>
+fn type_as_array(
+    arguments: Vec<(Value, Location)>,
+    return_type: Type,
+    location: Location,
+) -> IResult<Value> {
+    type_as(arguments, return_type, location, |typ| {
+        if let Type::Array(length, array_type) = typ {
+            Some(Value::Tuple(vec![Value::Type(*array_type), Value::Type(*length)]))
```

Comment:
Cool, this reversed arguments part  was the only part I was worried might be missed.

The only reason the order of length and the element type are switched in Type::Array is because several years ago arrays used to have a different syntax in Noir with the length first.

Reply:
> Cool, this reversed arguments part was the only part I was worried might be missed.

I actually wrote it in the wrong order the first time 😅 , and the assertion on the test file failed. I was 🤔 . Then I saw the order is reversed. Then I checked for the signature you suggested for `as_array` and made sure length was the second one 😄